src/main/java/tr/com/srdc/cda2fhir/transform/CCDTransformerImpl.java

```diff
@@ -137,24 +137,20 @@ public void setIdGenerator(IdGeneratorEnum idGen) {
      * @param resourceProfileMap The mappings of default resource profiles to desired resource profiles. Used to set profile URI's of bundle entries or omit unwanted entries.
      * @return A FHIR Bundle that contains a Composition corresponding to the CCD document and all other resources but Patient that are referenced within the Composition.
      */
-    public Bundle transformDocument(ClinicalDocument cda, BundleType bundleType, String patientRef, Map<String, String> resourceProfileMap) {
-        // The default transformer will use this patient reference if it is set.
-        this.patientRef = new Reference("Patient/" + patientRef);
-
-        Bundle documentBundle =  transformDocument(cda);
-        if (documentBundle == null) return null;
-
+    public Bundle createTransactionBundle(Bundle bundle, BundleType bundleType, Map<String, String> resourceProfileMap, boolean initialBundle) {
         Bundle resultBundle = new Bundle();
         resultBundle.setType(bundleType);
 
         switch (bundleType) {
             case TRANSACTION:
-                for(BundleEntryComponent entry : documentBundle.getEntry()) {
+                for(BundleEntryComponent entry : bundle.getEntry()) {
                     // Patient resource will not be added
-                    if (entry != null && !entry.getResource().getResourceType().name().equals("Patient")) {
+                    if (entry != null) {
```

Comment:
I think at some point we need to review external PatientRef and do not add patient resource to the bundle but that is not now I think.